src/com/company/AwardSystem.java

```diff
@@ -0,0 +1,30 @@
+package com.company;
+
+import java.util.HashMap;
+import java.util.Observable;
+import java.util.Observer;
+
+public class AwardSystem implements Observer {
+    private HashMap<Integer,Integer> awardset = new HashMap<Integer, Integer>();
+    private static volatile AwardSystem instance = new AwardSystem();
+    private   AwardSystem(){
+        awardset.put(1,5);
+        awardset.put(2,10);
+    }
+    public static AwardSystem getInstance() {
+        return instance;
+    }
+
+    @Override
+    public void update(Observable o, Object arg) {
```

Comment:
variable name o is not good. Later on in the code we would have to check the type, or look at method definition to see what o actually is.